src/DSharp.Compiler/Compiler/ExpressionBuilder.cs

```diff
@@ -1795,6 +1797,75 @@ private Expression ProcessOpenParenExpressionNode(BinaryExpressionNode node)
             return methodExpression;
         }
 
+        private (ExpressionListNode argumentNodes, List<Expression> argumentExpressions) ParseArguments(BinaryExpressionNode node, MemberExpression memberExpression)
+        {
+            ExpressionListNode argumentNodes = null;
+            List<Expression> argumentExpressions = new List<Expression>();
+
+            if (node.RightChild == null)
+            {
+                return (argumentNodes, argumentExpressions);
+            }
+
+            Debug.Assert(node.RightChild is ExpressionListNode);
+
+            if(memberExpression.Member is MethodSymbol methodSymbol && methodSymbol.IsGeneric)
+            {
+                GenericNameNode genericNameNode = GetGenericNameNode(node);
+                if (genericNameNode != null)
+                {
+                    var genericArgsMap = ParseTypeMap(methodSymbol, genericNameNode);
+                    if(genericArgsMap != null)
+                        argumentExpressions.Add(genericArgsMap);
+                }
+            }
+
+            argumentNodes = (ExpressionListNode)node.RightChild;
```

Comment:
node.RightChild needs to be an as ExpressionListNode unless we completely want to do this